apps/vmq_diversity/priv/auth/mysql.lua

```diff
@@ -0,0 +1,119 @@
+-- Copyright 2017 Erlio GmbH Basel Switzerland (http://erl.io)
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- MySQL Configuration, read the documentation below to properly
+-- provision your database.
+USER = "vernemq"
+PASSWORD = "vernemq"
+DATABASE = "vernemq_db"
+HOST = "127.0.0.1"
+PORT = 3306
+
+-- In order to use this Lua plugin you must deploy the following database
+-- schema and grant the user configured above with the required privileges:
+-- 
+-- CREATE TABLE vmq_auth_acl
+-- (
+--   mountpoint VARCHAR(10) NOT NULL,
+--   client_id VARCHAR(128) NOT NULL,
+--   username VARCHAR(128) NOT NULL,
+--   password VARCHAR(128),
+--   publish_acl TEXT,
+--   subscribe_acl TEXT,
+--   CONSTRAINT vmq_auth_acl_primary_key PRIMARY KEY (mountpoint, client_id, username)
+-- )
+--
+-- To insert a client ACL use a similar SQL statement:
+--
+-- INSERT INTO vmq_auth_acl 
+--   (mountpoint, client_id, username, 
+--    password, publish_acl, subscribe_acl)
+-- VALUES 
+--   ('', 'test-client', 'test-user', 
+--    PASSWORD('123'), '["a/b/c","c/b/#"]', '["a/b/c","c/b/#"]');
```

Comment:
The MySQL `PASSWORD()` function stores unsalted passwords. Shall we just accept that for MySQL as that's probably what people need or should we go for something more modern?